### PR TITLE
HPCC-16213 - Thor allocates 0 pages on OSX

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman rchapman@hpccsystems.com

### CODE REVIEW

**Comment by richardkchapman** [comment]
@jakesmith Please review

File: system/jlib/jdebug.cpp [hunk 1]
```diff
@@ -1103,8 +1105,37 @@ static unsigned evalAffinityCpus()
...
+    uint64_t inactiveBytes = (vmstat.inactive_count + vmstat.free_count) * pageSize;
+    uint64_t activeBytes = totalBytes - inactiveBytes;
+    uint64_t inuseBytes = (vmstat.wire_count + vmstat.active_count) * pageSize;
+
+    active = activeBytes / 1024;
+    inuse = inuseBytes / 1024;
+    total = totalBytes/1024;
+    swaptotal = 0;
+    swapinuse = 0;
+    DBGLOG("inuse=%u active=%u total=%u", inuse, active, total);
```
Comment (jakesmith) [inline system/jlib/jdebug.cpp]: accidentally left in I assume?

File: system/jlib/jdebug.cpp [hunk 2]
```diff
@@ -1103,8 +1105,37 @@ static unsigned evalAffinityCpus()
...
+// Note - values are returned in Kb
+
 static void getMemUsage(unsigned &inuse,unsigned &active,unsigned &total,unsigned &swaptotal,unsigned &swapinuse) 
 {
+#ifdef __APPLE__
+    int mib [] = { CTL_HW, HW_MEMSIZE };
+    int64_t totalBytes = 0;
+    size_t length = sizeof(totalBytes);
+
+    if(-1 == sysctl(mib, 2, &totalBytes, &length, NULL, 0))
```
Comment (jakesmith) [inline system/jlib/jdebug.cpp]: formatting: space after if

File: system/jlib/jdebug.cpp [hunk 3]
```diff
@@ -1103,8 +1105,37 @@ static unsigned evalAffinityCpus()
...
+        return;
+
+    vm_size_t pageSize;
+    if (KERN_SUCCESS != host_page_size(mach_host_self(), &pageSize))
+        return;
+    mach_msg_type_number_t count = HOST_VM_INFO_COUNT;
+    vm_statistics64_data_t vmstat;
+    if(KERN_SUCCESS != host_statistics64(mach_host_self(), HOST_VM_INFO, (host_info_t)&vmstat, &count))
```
Comment (jakesmith) [inline system/jlib/jdebug.cpp]: formatting: space after if

File: system/jlib/jdebug.cpp [hunk 4]
```diff
@@ -1103,8 +1105,37 @@ static unsigned evalAffinityCpus()
...
+    int mib [] = { CTL_HW, HW_MEMSIZE };
+    int64_t totalBytes = 0;
+    size_t length = sizeof(totalBytes);
+
+    if(-1 == sysctl(mib, 2, &totalBytes, &length, NULL, 0))
+        return;
+
+    vm_size_t pageSize;
+    if (KERN_SUCCESS != host_page_size(mach_host_self(), &pageSize))
+        return;
```
Comment (jakesmith) [inline system/jlib/jdebug.cpp]: Perhaps it should log and fill in values with 0's or similar?

File: system/jlib/jdebug.cpp [hunk 5]
```diff
@@ -1103,8 +1105,37 @@ static unsigned evalAffinityCpus()
...
+    if(KERN_SUCCESS != host_statistics64(mach_host_self(), HOST_VM_INFO, (host_info_t)&vmstat, &count))
+        return;
+
+    uint64_t inactiveBytes = (vmstat.inactive_count + vmstat.free_count) * pageSize;
+    uint64_t activeBytes = totalBytes - inactiveBytes;
+    uint64_t inuseBytes = (vmstat.wire_count + vmstat.active_count) * pageSize;
+
+    active = activeBytes / 1024;
+    inuse = inuseBytes / 1024;
+    total = totalBytes/1024;
```
Comment (jakesmith) [inline system/jlib/jdebug.cpp]: formatting/trivial: active/inuse lines space differently.

File: system/jlib/jdebug.cpp [hunk 6]
```diff
@@ -1103,8 +1105,37 @@ static unsigned evalAffinityCpus()
...
+    swaptotal = 0;
```
Comment (jakesmith) [inline system/jlib/jdebug.cpp]: minor: should there be a // MORE ? - it looks like you can get someswap stats via sysctlbyname(, e.g. sysctlbyname("vm.swapusage")

**Comment by jakesmith** [comment]
@richardkchapman - only a couple of minor comments

**Comment by jakesmith** [comment]
@richardkchapman - looks good.

**Comment by richardkchapman** [comment]
@ghalliday Please review/merge

**Pull request merged**